library/Vanilla/Theme/JsonAsset.php

```diff
@@ -41,9 +41,22 @@ public function asArray(): array {
     /**
      * Get the JSON content.
      *
-     * @return string
+     * @param string $type
+     * @return mixed
      */
-    public function getData(): string {
-        return $this->data;
+    public function getData(string $type = 'string') {
+        if ($type === 'array') {
+            if (is_array($this->data)) {
+                return $this->data;
+            } else {
+                return json_decode(json_encode($this->data, JSON_UNESCAPED_UNICODE), true);
+            }
+        }
+
+        if ($type === 'class') {
+            return (object)$this->data;
+        }
+
+        return json_encode($this->data, JSON_UNESCAPED_UNICODE);
```

Comment:
Can we leave this alone entirely and just a new method that would return the array?

We don't really care about the other use cases.

- `getData(): string` and keep your JSON encoding.
- `getVariables(): array` and just return the array directly.